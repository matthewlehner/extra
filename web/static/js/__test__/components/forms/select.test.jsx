import React from "react";
import renderer from "react-test-renderer";

import Select from "components/forms/select";

describe("Select component", () => {
  it("renders a select element", () => {
    const props = {
      value: "1",
      onChange: jest.fn()
    };

    const tree = renderer.create(
      <Select {...props}>
        <option value="1">Hi</option>
        <option value="2">Ho</option>
      </Select>
    );

    expect(tree).toMatchSnapshot();
  });
});
