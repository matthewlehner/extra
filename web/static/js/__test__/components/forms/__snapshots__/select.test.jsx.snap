// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component renders a select element 1`] = `
<div
  className="select"
>
  <select
    checkValidity={[Function]}
    onChange={[Function]}
    required={false}
    value="1"
  >
    <option
      value="1"
    >
      Hi
    </option>
    <option
      value="2"
    >
      Ho
    </option>
  </select>
</div>
`;
